frontend/packages/shared/src/components/altinnHeader/AltinnHeader.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { render as rtlRender, screen } from '@testing-library/react';
+import { AltinnHeader, AltinnHeaderProps } from './AltinnHeader';
+import {
+  ServicesContextProps,
+  ServicesContextProvider,
+} from 'app-development/common/ServiceContext';
+
+describe('AltinnHeader', () => {
+  it('should render AltinnHeaderMenu', () => {
+    render();
+    expect(screen.getByTestId('altinn-header-menu')).toBeInTheDocument();
+  });
+
+  it('should render AltinnHeaderButtons', () => {
+    render();
+    expect(screen.getByTestId('altinn-header-buttons')).toBeInTheDocument();
+  });
+
+  it('should render AltinnSubMenu when showSubMenu is true', () => {
+    render();
+    expect(screen.getByTestId('altinn-sub-menu')).toBeInTheDocument();
+  });
+
+  it('should not render AltinnSubMenu when showSubMenu is false', () => {
+    render({ showSubMenu: false });
+    expect(screen.queryByAltText('altinn-sub-menu')).not.toBeInTheDocument();
```

Comment:
Komponenten har aldri noe element med denne alt-teksten, så denne testen vil gi et positivt resultat også hvis du fjerner `not`. Bruk samme metode som i testen over, da tester vi at det faktisk er `showSubMenu` som påvirker resultatet.
```suggestion
    expect(screen.queryByTestId('altinn-sub-menu')).not.toBeInTheDocument();
```
Du kan også vurdere å mocke komponentene for å unngå å ha data-testid i DOM-en (se eksempel her: https://github.com/Altinn/altinn-studio/blob/master/frontend/packages/ux-editor/src/components/rightMenu/RightMenu.test.tsx), men det er på grensen til hva som egentlig er nødvendig. Man kan fint argumentere for at det er i orden å bruke test-id her.

Reply:
funget ikke med queryByTestId, funker bare med getByTestId

Reply:
Problemet er at `render` ikke tar i bruk `props`. Se egen kommentar.

Reply:
Fikset